app/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnector.scala

```diff
@@ -82,36 +92,48 @@ class EISConnectorImpl(
         Future.failed(new IllegalStateException(s"An unexpected error occurred - got a ${x.getClass}"))
     }
 
-  override def post(body: Source[ByteString, _], hc: HeaderCarrier): Future[Either[RoutingError, Unit]] =
+  private def extractHeaders(movementId: MovementId, messageId: MessageId, hc: HeaderCarrier): Seq[(String, String)] = {
+    val dateHeader: Seq[(String, String)] = hc.headers(Seq(HeaderNames.DATE)) match {
+      case Seq() => Seq(HeaderNames.DATE -> HTTP_DATE_FORMATTER.format(OffsetDateTime.now()))
+      case x     => x
+    }
+
+    val requestHeaders: Seq[(String, String)] = hc.headers(Seq("X-Message-Type")) ++ dateHeader ++ Seq(
+      "X-Correlation-Id"        -> UUID.randomUUID().toString,
```

Comment:
I think this  "X-Correlation-Id" and others below String better to define inside Constants.

Reply:
I'm somewhat open to this because this could benefit the tests too -- I may put these in a companion object rather than a constants object elsewhere though but they have no value outside of this connector.

Reply:
I've created a `RouterHeaderNames` class, but I've not used it in the tests -- as the tests that enforce a spec should be separate from the production code as much as possible, including constants.